src/items/FileItem.tsx

```diff
@@ -138,33 +140,37 @@ const FileItem: FC<FileItemProps> = ({
   }
 
   // todo: add more file extensions
-
+  let fileItem: JSX.Element;
   if (component) {
     // display element with caption
     if (showCaption) {
-      return withCaption({
+      fileItem = withCaption({
         item,
         onSave: onSaveCaption,
         edit: editCaption,
         saveButtonId,
       })(component);
+    } else {
+      fileItem = component;
     }
-
-    return component;
+  } else if (defaultItem) {
+    fileItem = defaultItem;
+  } else {
+    fileItem = (
+      <DownloadButtonFileItem
+        id={id}
+        name={originalFileName}
+        url={url}
+        text={downloadText}
+      />
+    );
   }
```

Comment:
This file was originally quite complex. 
Actually I would display the caption even tho there's a default or a download button.
Maybe since you have a complex if-else logic as well, I would refactor with a function like this (not sure how to call the function tho):
```javascript

const buildComponent = () => {

  if(mimetype) {
  // here mimetype switch
  return component;
  }
    if (defaultItem) {
    return defaultItem;
  }
  return <Download...>
}


fileItem = buildComponent( )

if(showCaption)...
if(showCollapse)...

```